src/Stratis.Bitcoin.Features.Wallet/Broadcasting/BroadcasterManagerBase.cs

```diff
@@ -57,9 +62,11 @@ public void AddOrUpdate(Transaction transaction, State state)
                 broadcastEntry.State = state;
                 this.OnTransactionStateChanged(broadcastEntry);
             }
+
+            this.walletManager.ProcessTransaction(transaction, null, null, state == State.Propagated);
```

Comment:
Looking at my comment inside `BroadcasterManager_TransactionStateChanged`,  I don't think you need the walletManager here, as the OnTransactionStateChanged method can probably be called every time we have a new transaction so yo udon't need to call ProcessTransaction from here.

Reply:
Hmm. That actually makes my last commit unnecessary since it was about adding wallet manager and fixing dependency loop that was caused by it. 
But yeah, if we just set IsPropagated = true for every incoming tx it makes sense to remove wallet manager from  BroadcasterManagerBase.cs completely.